spring-integration-ip/src/main/java/org/springframework/integration/ip/dsl/UdpOutboundChannelAdapterSpec.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.ip.dsl;
+
+import java.net.DatagramSocket;
+import java.util.function.Function;
+
+import org.springframework.integration.dsl.MessageHandlerSpec;
+import org.springframework.integration.expression.FunctionExpression;
+import org.springframework.integration.ip.udp.UnicastSendingMessageHandler;
+import org.springframework.messaging.Message;
+
+/**
+ * A {@link MessageHandlerSpec} for {@link UnicastSendingMessageHandler}s.
+ *
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public class UdpOutboundChannelAdapterSpec
+		extends MessageHandlerSpec<UdpOutboundChannelAdapterSpec, UnicastSendingMessageHandler> {
+
+	protected UdpOutboundChannelAdapterSpec() {
+		super();
+	}
+
+	UdpOutboundChannelAdapterSpec(String destinationExpression) {
+		this.target = new UnicastSendingMessageHandler(destinationExpression);
+	}
+
+	UdpOutboundChannelAdapterSpec(Function<Message<?>, ?> destinationFunction) {
```

Comment:
This ctor isn't covered in the `Udp`.
The same concern for the `UdpMulticastOutboundChannelAdapterSpec`.
Also I see a generic problem:
```
Udp.outboundMulticastAdapter("headers['udp_dest']")
					.lengthCheck(true)
					.timeToLive(10)
```
We can't reach an inheritors methods if we call the super.
So, this `UdpOutboundChannelAdapterSpec` must be with generic specification for itself. Exactly what you had before on the `TcpInboundChannelAdapterSpec` for example.
Or what we have on the `JmsOutboundChannelAdapterSpec` or `AbstractConnectionFactorySpec`.